ui-library-click-to-call/README.md

```diff
@@ -47,3 +53,1211 @@ Instead, it will copy all the configuration files and the transitive dependencie
+
+To do this, we will create some handlers to send post messages between the parent window and child window to signal that we want some information. See diagram:
+
+```mermaid
+graph LR
+  child[Child Window]
+  parent[Parent Window]
+  
+
+  parent --> |2. Adapter args| child
+  child -->|1. Args please| parent
+  
+```
+This flow illustrates that if the child window has spawned it needs to ask for the arguments. This has to do with React and that if the parent window just sends a message right after creation the call adapter arguments needed are lost when the application mounts. This is because in the new window the listener is not set yet until after a render pass completes. More on where these event handlers are made to come.
+
+Now we will want to update the splash screen we created earlier. First we will add a reference to the new child window that we will create.
+
+`ClickToCallScreen.tsx`
+
+```typescript
+    ...
+    const [userDisplayName, setUserDisplayName] = useState<string>();
+    const newWindowRef = React.useRef<Window | null>(null);
+    const [useVideo, setUseVideo] = useState<boolean>(false);
+    ...
+```
+
+Next we will create a handler that we pass to our widget that will create a new window that will start the process of sending the post messages.
+
+`ClickToCallScreen.tsx`
+```typescript
+    ...
+    const startNewWindow = useCallback(() => {
+        const startNewSessionString = 'newSession=true';
+        newWindowRef.current = window.open(
+            window.origin + `/?${startNewSessionString}`,
+            'call screen',
+            'width=500, height=450'
+        );
+    }, []);
+    ...
+```
+
+What this handler does is start a new window position and place a new query arg in the window URL so that the main application knows that it is time to start a new call.
+
+Next we will add a `useEffect` hook that is creating a event handler listening for new post messages from the child window.
+
+`ClickToCallScreen.tsx`
+```typescript
+    ...
+    useEffect(() => {
+        window.addEventListener('message', (event) => {
+            if (event.origin !== window.origin) {
+                return;
+            }
+            if (event.data === 'args please') {
+                const data = {
+                    userId: adapterParams.userId,
+                    displayName: adapterParams.displayName,
+                    token: adapterParams.token,
+                    locator: adapterParams.locator,
+                    alternateCallerId: adapterParams.alternateCallerId,
+                    useVideo: useVideo
+                };
+                console.log(data);
+                newWindowRef.current?.postMessage(data, window.origin);
+            }
+        });
+    }, [adapterParams, adapterParams.locator, adapterParams.displayName, useVideo]);
+    ...
+```
```

Comment:
This code needs to come after `const adapterParams` in the ClickToCallScreen.tsx